internal/pkg/agent/application/upgrade/artifact/download/errors/disk_space_errors_unix.go

```diff
@@ -0,0 +1,14 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```

Comment:
Nit: It is traditional to name files that cover all OSes except one with the `_other` suffix (instead of the `_unix` suffix here).